providers/openstack/zun.go

```diff
@@ -0,0 +1,509 @@
+
+// NodeConditions returns a list of conditions (Ready, OutOfDisk, etc), for updates to the node status
+// within Kubernetes.
+func (p *ZunProvider) NodeConditions(ctx context.Context) []v1.NodeCondition {
+	// TODO: Make these dynamic and augment with custom Zun specific conditions of interest
+	return []v1.NodeCondition{
+		{
+			Type:               "Ready",
+			Status:             v1.ConditionTrue,
+			LastHeartbeatTime:  metav1.Now(),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "KubeletReady",
+			Message:            "kubelet is ready.",
+		},
+		{
+			Type:               "OutOfDisk",
+			Status:             v1.ConditionFalse,
+			LastHeartbeatTime:  metav1.Now(),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "KubeletHasSufficientDisk",
+			Message:            "kubelet has sufficient disk space available",
+		},
+		{
+			Type:               "MemoryPressure",
+			Status:             v1.ConditionFalse,
+			LastHeartbeatTime:  metav1.Now(),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "KubeletHasSufficientMemory",
+			Message:            "kubelet has sufficient memory available",
+		},
+		{
+			Type:               "DiskPressure",
+			Status:             v1.ConditionFalse,
+			LastHeartbeatTime:  metav1.Now(),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "KubeletHasNoDiskPressure",
+			Message:            "kubelet has no disk pressure",
+		},
+		{
+			Type:               "NetworkUnavailable",
+			Status:             v1.ConditionFalse,
+			LastHeartbeatTime:  metav1.Now(),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "RouteCreated",
+			Message:            "RouteController created a route",
+		},
+	}
+}
+
+// NodeAddresses returns a list of addresses for the node status
+// within Kubernetes.
+func (p *ZunProvider) NodeAddresses(ctx context.Context) []v1.NodeAddress {
+	return nil
+}
+
+// NodeDaemonEndpoints returns NodeDaemonEndpoints for the node status
+// within Kubernetes.
+func (p *ZunProvider) NodeDaemonEndpoints(ctx context.Context) *v1.NodeDaemonEndpoints {
+	return &v1.NodeDaemonEndpoints{
+		KubeletEndpoint: v1.DaemonEndpoint{
+			Port: p.daemonEndpointPort,
+		},
+	}
+}
+
+// OperatingSystem returns the operating system for this provider.
+// This is a noop to default to Linux for now.
+func (p *ZunProvider) OperatingSystem() string {
+	return providers.OperatingSystemLinux
+}
+
+func capsuleToPod(capsule *capsules.Capsule) (*v1.Pod, error) {
+	var podCreationTimestamp metav1.Time
+	var containerStartTime metav1.Time
+
+	podCreationTimestamp = metav1.NewTime(capsule.CreatedAt)
+	if len(capsule.Containers) > 1 {
+		containerStartTime = metav1.NewTime(capsule.Containers[1].StartedAt)
+	}
+	containerStartTime = metav1.NewTime(time.Time{})
+	// Deal with container inside capsule
+	containers := make([]v1.Container, 0, len(capsule.Containers))
+	containerStatuses := make([]v1.ContainerStatus, 0, len(capsule.Containers))
+	// First container is sandbox
```

Comment:
This is not true anymore. The sandbox container is removed from the list. All the containers in the list are the actual containers declared in the pod.